### ISSUE TITLE
empty array with Array constructor

### ISSUE THREAD

**Issue by thecotne** [JSMonk/hegel#268]
```ts
const arr01: Array<number> = []
const arr02: Array<number> = new Array// should work
```

[try](https://hegel.js.org/try#MYewdgzgLgBAhgJwQBgIwC4YEElwJ4A8YArgLYBGApggHwwC8MA2gLoBQoksiKATJjgT4iZKrQYwwlAO7ZceAPQKYEABYhiAGwAmMaSAQBrNkA)

**Comment by JSMonk** [comment]
And we have an error, as I see in the playground. Or do I miss something?

**Comment by thecotne** [comment]
@JSMonk it should not report error as `new Array` and `[]` create exactly same value 

---

confusion comes from me editing `should work` into `should not work` because i misread and then when i realized it vas correct in first place i have changed back to `should work`

**Comment by JSMonk** [comment]
No, it should. Because after we will use the `Array` constructor, we will have empty values inside.
```ts
const arr: Array<number> = new Array<number>(3);
if (arr.length > 0) {
    const first = arr[0]; // upper value is valid, then we will have `undefined` inside instead of `number`;
}
```

**Comment by JSMonk** [comment]
`new Array` is the same as `Array<someType | undefined>`.

**Comment by thecotne** [comment]
array will have empty slots only if you pass single number to constructor and if number is greater than zero (if it's less then zero its a RangeError)

<img width="324" alt="Screen Shot 2020-06-10 at 10 09 53" src="https://user-images.githubusercontent.com/1606993/84233021-91a4c380-ab02-11ea-84d2-8dfed5d393ad.png">

implementing this to closely match javascript semantics might be over engineering at this point but it can be done
